crates/test/migration-test/tests/network_integration.rs

```diff
@@ -0,0 +1,403 @@
+/// Will return the most bonded, which means the Penumbra Labs CI validator.
+fn get_validator(tmpdir: &TempDir) -> String {
+    let mut validator_cmd = get_pcli_path();
+    validator_cmd
+        .args([
+            "--home",
+            tmpdir.path().to_str().unwrap(),
+            "query",
+            "validator",
+            "list",
+        ])
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    validator_cmd.assert().success();
+
+    // Pull out one of the validators from stdout.
+    let stdout_vec = validator_cmd.unwrap().stdout;
+    let validator_regex = Regex::new(r"penumbravalid1\w{58}").unwrap();
+    let captures = validator_regex.captures(std::str::from_utf8(&stdout_vec).unwrap());
+
+    // We retrieve the first match via index 0, which results in most trusted.
+    captures.unwrap()[0].to_string()
+}
+
+/// Look up current block height on target network.
+fn get_current_height(tmpdir: &TempDir) -> u64 {
+    let mut info_cmd = get_pcli_path();
+    info_cmd
+        .args([
+            "--home",
+            tmpdir.path().to_str().unwrap(),
+            "query",
+            "chain",
+            "info",
+        ])
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    info_cmd.assert().success();
+
+    let stdout_vec = info_cmd.unwrap().stdout;
+    let height_regex = Regex::new(r"Current Block Height\s+(\d+)").unwrap();
+    let captures = height_regex.captures(std::str::from_utf8(&stdout_vec).unwrap());
+
+    // Reminder: index 0 is entire match, index 1 is first captured group.
+    captures.unwrap()[1]
+        .parse()
+        .expect("block height in pcli output is u64")
+}
+
+// Needed functions:
+//
+//   get_current_height()
+//   generate_proposal
+//   submit_proposal
+//
+//
+
+/// Look up top validator and delegate `TEST_DELEGATION` to it.
+#[cfg_attr(not(feature = "migration-test"), ignore)]
+#[test]
+fn delegate() {
+    tracing_subscriber::fmt::try_init().ok();
+    tracing::info!("delegate_and_undelegate");
+    let tmpdir = load_wallet_into_tmpdir();
+
+    // Get a validator from the testnet.
+    let validator = get_validator(&tmpdir);
+
+    // Delegate a tiny bit of penumbra to the validator.
+    let mut delegate_cmd = get_pcli_path();
+    delegate_cmd
+        .args([
+            "--home",
+            tmpdir.path().to_str().unwrap(),
+            "tx",
+            "delegate",
+            &TEST_DELEGATION,
+            "--to",
+            validator.as_str(),
+            // use dedicated subaccount for smoke delegations
+            "--source",
+            "2",
+        ])
+        .timeout(std::time::Duration::from_secs(TIMEOUT_COMMAND_SECONDS));
+    tracing::info!(?delegate_cmd, "running delegation command");
+    delegate_cmd.assert().success();
+    // Wait for bonding, otherwise delegation isn't active.
+    tracing::info!("waiting for stake to bonding...");
+    thread::sleep(*EPOCH_DURATION);
+}
```

Comment:
:grimacing: is this depending on cargo test running tests in a certain order? that sounds like a recipe for very confusing errors in the future, i'm very hesitant to follow that course if so. if another test is dependent on side-effects of this test first, we should consolidate them into a single test.

Reply:
since we aren't making any assertions about unbonding, we can save time by not waiting around.

```suggestion
}
```